### PR TITLE
Add a note on `:jekyll_plugins` group in the docs

### PR DESCRIPTION
**Pull request by ashmaroli**
Ref: https://github.com/jekyll/jekyll/issues/6358#issuecomment-339859914

gem-group

/cc @jekyll/documentation

### CODE REVIEW

**Comment by DirtyF** [comment]
Could we recommend this as the default ? I mean you just list your gems in :jekyll_plugins and you're done. IIUC only GitHub Pages users have to rely on the `plugins` key in the `_config.yml`

**Comment by ashmaroli** [comment]
> should we recommend this as the default

Since the documentation already states that these three options are equally potent and intermixable, I don't think *one of those options* can be a default..

File: docs/_docs/plugins.md [hunk 1]
```diff
@@ -58,6 +58,21 @@ You have 3 options for installing plugins:
 
 <div class="note info">
   <h5>
+    The <code>:jekyll_plugins</code> group in a <code>Gemfile</code>
+  </h5>
+  <p>
+    Jekyll treats this particular group of gems(s) with a special love. Any gem
```
Comment (mattr-) [inline docs/_docs/plugins.md]:
```
Jekyll gives this particular group of gems different treatment.
```

instead of
```
Jekyll treats this particular group of gems(s) with a special love.
```

File: docs/_docs/plugins.md [hunk 2]
```diff
@@ -58,6 +58,21 @@ You have 3 options for installing plugins:
...
+    included in this group is given a high priority and "loaded" way before
```
Comment (mattr-) [inline docs/_docs/plugins.md]: ~given a high priority and~ 

remove the quotes around `loaded`.

Remove the word `way` from this line.

File: docs/_docs/plugins.md [hunk 3]
```diff
@@ -58,6 +58,21 @@ You have 3 options for installing plugins:
...
+    Jekyll starts processing the rest of your source directory.
+    <br/>
+    Because of the priority, a gem included here will be activated (even under
```
Comment (mattr-) [inline docs/_docs/plugins.md]: ~Because of the priority,~ 

Capitalize the a after removing the priority bit.

File: docs/_docs/plugins.md [hunk 4]
```diff
@@ -58,6 +58,21 @@ You have 3 options for installing plugins:
...
+    <code>--safe</code> mode), even if its not explicitly listed under the
```
Comment (mattr-) [inline docs/_docs/plugins.md]: This `--safe` note is important. It should get at least it's own sentence and perhaps even a warning box wrapper as well. Something like

```
Gems included in the `jekyll-plugins` group are activated regardless of the `--safe` mode setting. Be careful what gems are included here!
```

File: docs/_docs/plugins.md [hunk 5]
```diff
@@ -58,6 +58,21 @@ You have 3 options for installing plugins:
 
 <div class="note info">
   <h5>
+    The <code>:jekyll_plugins</code> group in a <code>Gemfile</code>
```
Comment (mattr-) [inline docs/_docs/plugins.md]: I think I'd prefer if this is not a note and just an added section to the documentation.

**Comment by mattr-** [comment]
> As we require Bundler, should we recommend this as the default?

We should probably document the order of precedence each one has rather than recommending a default and just leave it at that.

File: docs/_docs/plugins.md (hunk 1 above)
Comment (ashmaroli) [reply docs/_docs/plugins.md]: or how about
```
Jekyll treats this particular group of gems(s) differently
```
?

**Comment by ashmaroli** [comment]
> I have a few things I'd like to see addressed

...done.

File: docs/_docs/plugins.md (hunk 1 above)
Comment (mattr-) [reply docs/_docs/plugins.md]: That's fine too. Just need one `s` on `gem(s)` though. 😄

File: docs/_docs/plugins.md [hunk 6]
```diff
@@ -83,6 +68,24 @@ You have 3 options for installing plugins:
   </p>
 </div>
 
+### The jekyll-plugins group
+
+Jekyll gives this particular group of gems in your `Gemfile`, a different
```
Comment (mattr-) [inline docs/_docs/plugins.md]: no comma necessary.

**Comment by ashmaroli** [comment]
Thanks Matt :smiley:

File: docs/_docs/plugins.md [hunk 7]
```diff
@@ -68,6 +68,24 @@ You have 3 options for installing plugins:
   </p>
 </div>
 
+### The jekyll-plugins group
```
Comment (DirtyF) [inline docs/_docs/plugins.md]: s/jekyll_plugins

**Comment by DirtyF** [comment]
@jekyllbot: merge +docs

File: docs/_docs/plugins.md [hunk 8]
```diff
@@ -68,6 +68,24 @@ You have 3 options for installing plugins:
...
+processing the rest of your source directory.
+
+A gem included here will be activated even if its not explicitly listed under
+the `plugins:` key in your site's config file.
+
+<div class="note warning">
+  <p>
+    Gems included in the <code>:jekyll-plugins</code> group are activated
+    regardless of the <code>--safe</code> mode setting. Be careful of what
+    gems are included under this group!
```
Comment (parkr) [inline docs/_docs/plugins.md]: I'm a little weary of the FUD (fear, uncertainty, and doubt) that "Be careful!" leads to. Could we rephrase this to be less fear-mongery? "Be aware" perhaps?

Comment (DirtyF) [reply docs/_docs/plugins.md]: oops didn't see you comment, but totally agree with you. Will fix this right away

Comment (DirtyF) [reply docs/_docs/plugins.md]: fixed in https://github.com/jekyll/jekyll/commit/52c34060e3311c1d2a369f02b4d2319c1b283f7b

Comment (mattr-) [reply docs/_docs/plugins.md]: I didn't even notice this. 🙈  Thanks for pointing it out @parkr.
